interface.go

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package messaging
+
+// MessageClient is the messaging interface for publisher-subsriber pattern
+type MessageClient interface {
+	// Connect to messaging host specified in MessageBusconfig
+	// returns error if not able to connect
+	Connect(msgBusConfig *MessageBusConfig) error
```

Comment:
remove msgBusConfig *MessageBusConfig. This will be passed to the factory.NewMessageClient